server/src/main/java/org/elasticsearch/common/logging/JsonThrowablePatternConverter.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache license, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the license for the specific language governing permissions and
+ * limitations under the license.
+ */
+package org.elasticsearch.common.logging;
+
+import com.fasterxml.jackson.core.io.JsonStringEncoder;
+import org.apache.logging.log4j.core.LogEvent;
+import org.apache.logging.log4j.core.config.Configuration;
+import org.apache.logging.log4j.core.config.plugins.Plugin;
+import org.apache.logging.log4j.core.pattern.ConverterKeys;
+import org.apache.logging.log4j.core.pattern.ExtendedThrowablePatternConverter;
+import org.apache.logging.log4j.core.pattern.PatternConverter;
+import org.apache.logging.log4j.core.pattern.ThrowablePatternConverter;
+import org.apache.logging.log4j.util.Strings;
+
+import java.nio.charset.Charset;
+import java.util.StringJoiner;
+
+/**
+
+ * Outputs the Throwable portion of the LoggingEvent as a Json formatted field with array
+ * "exception": [ "stacktrace", "lines", "as", "array", "elements" ]
+ *
+ * Reusing @link org.apache.logging.log4j.core.pattern.ExtendedThrowablePatternConverter which already converts a Throwable from
+ * LoggingEvent into a multiline string
+ */
+@Plugin(name = "JsonThrowablePatternConverter", category = PatternConverter.CATEGORY)
+@ConverterKeys({"exceptionAsJson"})
+public final class JsonThrowablePatternConverter extends ThrowablePatternConverter {
+    private static final JsonStringEncoder JSON_STRING_ENCODER = JsonStringEncoder.getInstance();
```

Comment:
I checked the implementation of `JsonStringEncoder`. It never mentions it explicitly but it does not appear to be thread-safe (given that `#getInstance()` maintains a thread-local). So I think it is not safe to store it as a field.

Reply:
great point, I didn't expect that. will refactor into local use